src/RowGateway/AbstractRowGateway.php

```diff
@@ -204,7 +204,7 @@ public function delete()
         $where = [];
         // primary key is always an array even if its a single column
         foreach ($this->primaryKeyColumn as $pkColumn) {
-            $where[$pkColumn] = $this->primaryKeyData[$pkColumn];
+            $where[$pkColumn] = ($this->primaryKeyData ? $this->primaryKeyData[$pkColumn] : null);
```

Comment:
two comments here:
- brackets are redundant
- we should use `isset($this->primaryKeyData[$pkColumn])`, imo